python_modules/dagster/dagster/_core/storage/cloud_storage_compute_log_manager.py

```diff
@@ -48,6 +48,16 @@ def delete_logs(
     def download_url_for_type(self, log_key: Sequence[str], io_type: ComputeIOType) -> str:
         """Calculates a download url given a log key and compute io type."""
 
+    @abstractmethod
+    def uri_or_path_for_type(self, log_key: Sequence[str], io_type: ComputeIOType) -> Optional[str]:
+        """Calculates a download uri given a log key and compute io type."""
+        return None
```

Comment:
raise instead of return none

Reply:
I don't think it exactly fits the purview of an abstractmethod then right? Perhaps remove that decorator if you want to have a base implementation of None

Reply:
An abstractmethod would imply that every subclass needs to implement the method, otherwise it's still considered an "abstract class" which cannot be instantiated